### PR TITLE
Ad-hoc fix for `ref readonly` parameters

### PR DESCRIPTION
**Pull request by vzarytovskii**
Fixes https://github.com/dotnet/runtime/issues/94317

**Note that this is an ad-hoc fix which will just treat `ref readonly` as `inref` during parameters import, for long-term fix, please refer to https://github.com/dotnet/fsharp/issues/16222**

`ref readonly` is encoded as `[in]&T` with `RequiresLocationAttribute`, we import `inref<_>` only from `[in]&T` with `IsReadOnlyAttribute`, this PR adds a second check.

### CODE REVIEW

**Comment by vzarytovskii** [comment]
That's ready @dotnet/fsharp-team-msft 

@baronfel what should we do to backport it to `8.0.1xx` which will follow the release? We probably don't want to wait until `8.0.2xx`.

If 17.8 maps to `8.0.1xx` now in darc, should we just make a back port after release?

**Comment by baronfel** [comment]
Yeah - backport to the branch you have that maps to release/8.0.1xx in the SDK, then get Tactics approval before you merge that PR. Codeflow via darc will take care of the rest and you'll be automatically included in the first servicing release (in December).

**Review by abonie — APPROVED** [review]

**Comment by tannergooding** [comment]
> ref readonly is encoded as [in]&T with RequiresLocationAttribute, we import inref<_> only from [in]&T with IsReadOnlyAttribute, this PR adds a second check.

@vzarytovskii, it's worth noting that it while Roslyn currently encodes it as `[in]&T`, it is not required to have the `[in]` and only actually looks for `&T` + `RequiresLocation` or `[IsReadOnly]`.

It looks like F# isn't actually checking for `[in]` itself, but wanted to make the callout on what's part of the requirement vs not.

**Comment by vzarytovskii** [comment]
> > ref readonly is encoded as [in]&T with RequiresLocationAttribute, we import inref<_> only from [in]&T with IsReadOnlyAttribute, this PR adds a second check.
> 
> 
> 
> @vzarytovskii, it's worth noting that it while Roslyn currently encodes it as `[in]&T`, it is not required to have the `[in]` and only actually looks for `&T` + `RequiresLocation` or `[IsReadOnly]`.
> 
> 
> 
> It looks like F# isn't actually checking for `[in]` itself, but wanted to make the callout on what's part of the requirement vs not.

Yeah, I think we only emit it, not read. As you said, we pretty much only check attributes for refs

**Pull request merged**
